### ISSUE TITLE
Support `.keras` format in SWI for complied model to continue re-training

### ISSUE THREAD

**Issue by innat** [innat/medic-ai#101]
**Use case:**

To continue or resume training, it is required to ensure to cover up all compiled config for the re-trainable model. That includes loss, metric function. Currently some of the loss and metric have minimal get-config implementation. This hinders the scope of loading full compiled model for retraining.

The inference utility `SlidingWindowInference` only support `.h5` format. This should also support `.keras` format which would enable to support full model in training time.
